ms2deepscore/models/SiameseSpectralModel.py

```diff
@@ -132,59 +94,43 @@ def output_size(self):
 
 class SpectralEncoder(nn.Module):
     def __init__(self,
-                 base_dims,
-                 embedding_dim,
-                 dropout_rate,
-                 train_binning_layer: bool, group_size: int, output_per_group: int,
+                 model_settings: SettingsMS2Deepscore,
                  peak_inputs: int,
                  additional_inputs: int,
                  ):
         """
         Parameters
         ----------
-        base_dims
-            Tuple of integers depicting the dimensions of the desired hidden
-            layers of the base model
-        embedding_dim
-            Dimension of the embedding (i.e. the output of the base model)
-        train_binning_layer
-            Default is True in which case the model contains a first dense multi-group peak binning layer.
-        group_size
-            When binning layer is used the group_size determines how many input bins are taken into
-            one dense micro-network.
-        output_per_group
-            This sets the number of next layer bins each group_size group of inputs shares.
-        dropout_rate
-            Dropout rate to be used in the base model.
+        model_settings:
+            SettingsMS2Deepscore object storing all the settings
```

Comment:
I think that makes it hard to use for others.

We now put everything together in one huge settings class. This is nice for our current scripts because we can pass the same object to all functions.
But it also somewhat hides the main parameters I think.

I would probably leave it as you suggest for now. But let's add an issue to reconsider the general parameter handling. For instance, it might not be clear to people using it why specific generator settings are also running under model parameters now etc.

Reply:
The last changes I made were merging the GeneratorSettings with the Model_settings, so it is quite easy to rollback now. I would be fine with that as well.

Reply:
But I am not sure how confused the user would be. I think renaming the SettingsMS2Deepscore and model_settings to TrainingSettings would help a lot (like you suggested below). In that case, it is a nice overview of all the settings used. Having it split with the GeneratorSettings also still works and might make it a bit easier to interpret from just the parameter name what is meant, since it is already clear that this is about data generation.